arm-customer-insights/2017-04-26/swagger/customer-insights.json

```diff
@@ -0,0 +1,5664 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.CustomerInsights/hubs": {
+      "get": {
+        "tags": [
+          "Hubs"
+        ],
+        "operationId": "Hubs_List",
+        "x-ms-examples": {
+          "Hubs_List": {
+            "$ref": "../examples/HubsList.json"
+          }
+        },
+        "description": "Gets all hubs in the specified subscription.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Successfully get all the hubs in the subscription.",
+            "schema": {
+              "$ref": "#/definitions/HubListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/profiles/{profileName}": {
+      "put": {
+        "tags": [
+          "Profiles"
+        ],
+        "operationId": "Profiles_CreateOrUpdate",
+        "x-ms-examples": {
+          "Profiles_CreateOrUpdate": {
+            "$ref": "../examples/ProfilesCreateOrUpdate.json"
+          }
+        },
+        "description": "Creates a profile within a Hub, or updates an existing profile.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "hubName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the hub."
+          },
+          {
+            "name": "profileName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the profile.",
+            "pattern": "^[a-zA-Z][a-zA-Z0-9_]+$",
+            "minLength": 1,
+            "maxLength": 128
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ProfileResourceFormat"
+            },
+            "description": "Parameters supplied to the create/delete Profile type operation"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Successfully created the profile.",
+            "schema": {
+              "$ref": "#/definitions/ProfileResourceFormat"
+            }
+          },
+          "202": {
```

Comment:
hi, I don't know any specific reason except that this is how the API was designed and implemented. At this point we're not planning to make change to this API.